File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.py
# Last updated: 06/07/2026, 19:05:58
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if (str1 + str2 == str2 + str1):
            ind = gcd(len(str1), len(str2))
            return str1[:ind]
        return ""

File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.cpp
// Last updated: 06/07/2026, 19:05:52
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if (str1 + str2 == str2 + str1){
            int index = gcd(str1.size(), str2.size());
            return str1.substr(0, index);
        }
        return "";
    }
};